pkgs/development/python-modules/dalle-mini/default.nix

```
@@ -0,0 +1,48 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, einops
+, emoji
+, flax
+, ftfy
+, jaxlib
+, pillow
+, transformers
+, unidecode
+, wandb
+}:
+
+buildPythonPackage rec {
+  pname = "dalle-mini";
+  version = "0.1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-Sbos44uWGnJLYMx/xy0wkyAJHlDhVIeOS7rnYt2W53w=";
+  };
+
+  format = "setuptools";
+
+  propagatedBuildInputs = [
+    einops
+    emoji
+    flax
+    ftfy
+    jaxlib
```

```suggestion
    jax
```
according to `setup.cfg`. I wonder why it doesn't complain.

---

Hmm, setup only requires jax but anything using flax needs to use jaxlib for optax:

```
INFO: pip is looking at multiple versions of dalle-mini to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement jaxlib>=0.1.37 (from optax) (from versions: none)
ERROR: No matching distribution found for jaxlib>=0.1.37
```

So I guess I'll specify both?

---

https://nixos.wiki/wiki/JAX#How_do_I_package_JAX_libraries.3F

---

Thanks, wasn't aware of that